portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,38 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private String convertToJson(ArrayList<String> comments) {
+    Gson gson = new Gson();
+    String json = gson.toJson(comments);
+    return json;
+  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Vincent!");
+    
+    ArrayList<String> messages = new ArrayList<String>();
+    messages.add("Hello");
+    messages.add("This is a message.");
+    messages.add("I am a machine...");
+    
+    String json = convertToJson(messages);
+    
+    response.setContentType("application/json;");
```

Comment:
can you factor out this string ("application/json") as a static constant in the class?

private static final String CONTENT_TYPE = "application/json;";

Reply:
How would that be done?

Reply:
You can add the constant to the class by adding:

private static final String CONTENT_TYPE = "application/json;";

below the name of the class.

Then you can replace the string literal on line 44 with CONTENT_TYPE

Reply:
It looks like you added the constant CONTENT_TYPE, but you are still using a string literal here. Can you remove this string and replace it with the CONTENT_TYPE variable.